# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsGlobalWindowInner.h
# Commit: 4ccb6970887e
# Full Hash: 4ccb6970887ece212a1176681a567480ea586dfc
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2025-02-25 09:30:56
# Regressor Bug: 1920115
# File Overlap Count: 2
# Description:
#   Bug 1920115 - Allow high priority tasks to run before timers r=smaug
#   
#   In Gecko, unlike other runnables where the TaskController
#   picks one each at a time, the TimeoutManager allows multiple
#   timers to run sequentially without allowing other runnables
# ==============================================================================

diff -r 7522970c5a1d -r 4ccb6970887e dom/base/nsGlobalWindowInner.h
--- a/dom/base/nsGlobalWindowInner.h	Mon Feb 24 19:31:28 2025 +0000
+++ b/dom/base/nsGlobalWindowInner.h	Mon Feb 24 19:31:28 2025 +0000
@@ -338,6 +338,7 @@
   virtual bool HasActiveIndexedDBDatabases() override;
   virtual bool HasActivePeerConnections() override;
   virtual bool HasOpenWebSockets() const override;
+  virtual bool HasScheduledNormalOrHighPriorityWebTasks() const override;
   void SyncStateFromParentWindow();
 
   // Called on the current inner window of a browsing context when its